src/components/compound/DlCharts/charts/DlConfusionMatrix/DlConfusionMatrix.vue

```diff
@@ -0,0 +1,520 @@
+                </div>
+                <dl-brush
+                    track-size="18px"
+                    thumb-size="20px"
+                    :max="matrix.length"
+                    :max-range="2"
+                    :selection-color="getCellBackground(0.1)"
+                    :color="getCellBackground()"
+                    :step="1"
+                    drag-range
+                    @update:model-value="handleBrushUpdate"
+                />
+                <span class="label-tag x"> {{ bottomLabel }} </span>
+            </div>
+            <div
+                ref="colorSpectrum"
+                class="color-spectrum"
+            >
+                <div class="color-spectrum__gradient" />
+                <div class="color-spectrum__gradation">
+                    <div
+                        v-for="value in gradationValues"
+                        :key="value"
+                        class="color-spectrum__gradation--element"
+                    >
+                        <span class="color-spectrum__gradation--element-line">-</span>
+                        {{ value !== gradationValues[0] ? value : null }}
+                    </div>
+                </div>
+            </div>
+        </div>
+        <div
+            v-else
+            class="invalid"
+        >
+            The given props cannot create a valid matrix.
+        </div>
+        <div
+            v-if="tooltipState?.visible"
+            :style="tooltipStyles"
+            class="tooltip"
+        >
+            <span class="tooltip__x">{{ tooltipState.xLabel }}</span>
+            <span class="tooltip__y">{{ tooltipState.yLabel }}</span>
+            <span>
+                <span
+                    v-if="tooltipState.value"
+                    class="tooltip__color"
+                    :style="`background-color: ${getCellBackground(
+                        tooltipState.value
+                    )}`"
+                />
+                {{ normalized ? 'Normalized' : 'Unnormalized' }}
+                {{ tooltipState.value }}
+            </span>
+        </div>
+    </div>
+</template>
+
+<script lang="ts">
+import { defineComponent, PropType } from 'vue-demi'
+import DlBrush from '../../components/DlBrush.vue'
+import DlTooltip from '../../../../essential/DlTooltip/DlTooltip.vue'
+import { MatrixCell, Label, BrushState } from './types'
+import { hexToRgbA } from '../../../../../utils/colors'
+import { colorNames } from '../../../../../utils/css-color-names'
+import { useThemeVariables } from '../../../../../hooks/use-theme'
+import {
+    getGradationValues,
+    normalizeMatrix,
+    validateMatrix,
+    setZoom,
+    getCellWidth
+} from './utils'
+import { debounce } from 'lodash'
+export default defineComponent({
+    components: {
+        DlBrush,
+        DlTooltip
+    },
+    props: {
+        labels: {
+            required: true,
+            type: Array as PropType<string[] | Label[]>,
+            default: (): [] => []
```

Comment:
add for typing as string[] | label[]

Reply:
the default is returned an array of any which causes typing instability for some reason. so add 
```(): string[] | label[] => []```